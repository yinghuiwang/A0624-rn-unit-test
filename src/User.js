import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types'

class User extends React.Component {
  static propTypes = {
    abc: PropTypes.number.isRequired,
    b: PropTypes.number,
    onPress: PropTypes.func
  }

  state = {
    name: 'wangyinghui',
  }

  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <Text>{this.props.abc}</Text>
        <Text>{this.props.abc}</Text>
        <Text>{this.state.name}</Text>
      </TouchableOpacity>
    )
  }
}

export default User;