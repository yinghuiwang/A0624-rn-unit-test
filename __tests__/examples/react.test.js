import 'react-native'
import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import User from '../../src/User';

test('<User/>', () => {
  const component = renderer.create(
    <User
      abc={1111}
      b={2}
      onPress={() => {
        console.log('click')
      }}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // console.log(component.state.name);
  // expect(component.props.a).toBe(1);s
});

test('<User/> enzyme', () => {
  let i = 0
  const onPress = () => i++
  const wrapper = shallow(<User abc={1} b={2} onPress={onPress}/>);
  expect(wrapper.instance().props.abc).toBe(1); // uses the right handler
  expect(wrapper.prop('onPress')).toBe(onPress)
  expect(i).toBe(0);
  wrapper.simulate('press');
  expect(i).toBe(1);

});