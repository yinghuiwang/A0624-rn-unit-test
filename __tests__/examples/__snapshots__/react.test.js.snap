// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User/> 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    1111
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    1111
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    wangyinghui
  </Text>
</View>
`;
